presto-docs/src/main/sphinx/admin/dynamic-filtering.rst

```diff
@@ -0,0 +1,219 @@
+=================
+Dynamic Filtering
+=================
+
+Dynamic filtering optimizations significantly improve the performance of queries
+with selective joins by avoiding reading of data that would be filtered by join condition.
+Consider the following query which captures a common pattern of a fact table ``store_sales``
+joined with a filtered dimension table ``date_dim``:
+
+    SELECT COUNT(*) FROM
+    store_sales JOIN date_dim ON store_sales.ss_sold_date_sk = date_dim.d_date_sk
+    WHERE d_following_holiday='Y' AND d_year = 2000;
+
+Without dynamic filtering, Presto will push predicates for the dimension table to the
+table scan on ``date_dim`` but it will scan all the data in the fact table since there
+are no filters on ``store_sales`` in the query. The join operator will end up throwing away
+most of the probe-side rows as the join criteria is highly selective.
+
+When dynamic filtering is enabled, Presto will collect candidate values for join condition
+from the processed dimension table on the right side of join. In the case of broadcast joins,
+the runtime predicates generated from this collection are pushed into the local table scan
+on the left side of join running on the same worker.
+Additionally, these runtime predicates are communicated to the coordinator over the network
+so that dynamic filtering can also be performed on the coordinator during enumeration of
+table scan splits. For example, in the case of Hive connector, dynamic filters are used
+to skip loading of partitions which don't match the join criteria.
+This is known as dynamic partition pruning.
+
+The results of dynamic filtering optimization can include the following benefits:
+
+* improved overall query performance
+* reduced network traffic between Presto and the data source
+* reduced load on the remote data source
+
+Support for push down of dynamic filters is specific to each connector
+and the relevant underlying database or storage system.
+
+Analysis and Confirmation
+-------------------------
+
+Dynamic filtering depends on a number of factors:
+
+* Planner support for dynamic filtering for a given join operation in Presto.
+  Currently inner joins with equality join condition and semi-joins with IN conditions are supported.
+* Connector support for utilizing dynamic filters pushed into table scan at runtime.
+  For example, Hive connector can push dynamic filters into ORC and Parquet readers
+  to perform stripe or row-group pruning.
+* Connector support for utilizing dynamic filters at the splits enumeration stage.
+* Size of right (build) side of join.
+
+To analyze if the planner is adding dynamic filters to a specific query's plan,
```

Comment:
Turn the sentence around ..

You can take a closer look at the ... to analyze if the planner..